modules/test-operators/helpers/database.go

```diff
@@ -67,7 +83,17 @@ func (tc *TestHelper) DeleteDBService(name types.NamespacedName) {
 	}, tc.Timeout, tc.Interval).Should(t.Succeed())
 }
 
-// GetMariaDBDatabase -
+// GetMariaDBDatabase - retrieves a MariaDBDatabase resource from the Kubernetes cluster.
+//
+// The function creates a new MariaDBDatabase pointer and uses the Get function of the Kubernetes client to fetch the MariaDBDatabase.
+// If the MariaDBDatabase cannot be immediately retrieved, the function will retry for a duration specified by tc.Timeout,
+// checking every tc.Interval, until the MariaDBDatabase is successfully retrieved or the timeout duration is reached.
+// The function asserts that the retrieval operation should eventually succeed.
+//
+// The function returns a pointer to the retrieved MariaDBDatabase.
```

Comment:
This also feel chatty to me and very implementation specific. Also repeating things that already clear from the signature, like the return type.

For me the following would be enough:

GetMariaDBDatabase waits for and retrieves a MariaDBDatabase resource from the Kubernetes cluster